lib/service/database.js

```diff
@@ -294,7 +298,8 @@ DatabaseService.prototype.queueAnalysisOperations = function(analysis, callback)
                     id: asyncQueryId(analysis, nodeToUpdate),
                     query: transactionQuery([
                         deleteFromCacheTableQuery(targetTableName),
-                        populateCacheTableQuery(targetTableName, nodeToUpdate.sql())
+                        populateCacheTableQuery(targetTableName, nodeToUpdate.sql()),
+                        checkCacheTableQuery(targetTableName)
```

Comment:
Currently, since we do the same check for every call of `checkCacheTableQuery` the actual difference would be that we would check before the first cache table and not after the last. But if we get to use the `targetTableName` argument, would we want to have the table just created or the table to be created?

To cover all bases we could have two functions and:
```
deleteFromCacheTableQuery(targetTableName),
preCheckCacheTableQuery(targetTableName),
populateCacheTableQuery(targetTableName, nodeToUpdate.sql()),
postCheckCacheTableQuery(targetTableName)
```

And do in the `preCheck...` what we are doing now.

But I'm not sure, maybe should keep this simple, just check before as you suggest to fail as soon and possible and we'll consider more complex approaches [elsewhere](https://github.com/CartoDB/camshaft/issues/209)